### PR TITLE
judge: add type annotations

### PR DESCRIPTION
**Pull request by criyle**
- Add type annotations for sandbox client
- Add type annotation for most of functions
- Fixed typo destroy and acquire
- Skip `mount` for non-root user

### CODE REVIEW

File: packages/hydrojudge/src/sandbox.ts [hunk 1]
```diff
@@ -1,50 +1,79 @@
...
+}
+
+function checkStringArray(args: ParseEntry[]): args is string[] {
+    return args.every((arg: ParseEntry) => typeof arg === 'string');
+}
+
+function parseArgs(execute: string): string[] {
+    const args = cmd(execute.replace(/\$\{dir\}/g, '/w'));
+    if (!checkStringArray(args)) {
+        throw new FormatError(`${execute} contains invalid operator`);
```
Comment (undefined-moe) [inline packages/hydrojudge/src/sandbox.ts]: FormatError means wrong testdata or configuration format, this is system language definition issue and SystemError should be thrown.

File: packages/hydrojudge/src/sandbox_interface.ts [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    pipe?: boolean; // collect over pipe or not (default false)
+}
+
+export type CopyInFile = LocalFile | MemoryFile | PreparedFile;
+
+export interface Cmd {
+    args: string[]; // command line argument
+    env?: string[]; // environment
+
+    // specifies file input / pipe collector for program file descriptors
```
Comment (undefined-moe) [inline packages/hydrojudge/src/sandbox_interface.ts]: Field description should be in jsdoc style.

**Review by undefined-moe — CHANGES_REQUESTED** [review]

File: packages/hydrojudge/src/judge/index.ts [hunk 3]
```diff
@@ -1,9 +1,10 @@
...
 import * as hack from './hack';
 import * as interactive from './interactive';
+import { Context } from './interface';
 import * as run from './run';
 import * as submit_answer from './submit_answer';
 
 export = {
     default: def, interactive, run, submit_answer, hack, objective: submit_answer,
-};
+} as { [key: string]: { judge(ctx: Context): Promise<void> } };
```
Comment (undefined-moe) [inline packages/hydrojudge/src/judge/index.ts]:
```suggestion
} as Record<string, { judge(ctx: Context): Promise<void> }>;
```

**Comment by undefined-moe** [comment]
是不是把 client 和相关 types 独立发布到另一个 npm 包会更加合理（

**Comment by criyle** [comment]
> 是不是把 client 和相关 types 独立发布到另一个 npm 包会更加合理（

有亿点麻烦，目前有很多强耦合。之后有时间重构再考虑下🤔

**Pull request merged**